debian/bin/restart.sh

```diff
@@ -11,23 +11,42 @@
 #
 # This script is used by the Debian/Ubuntu/Fedora/CentOS packages.
 
-get_service_command() {
-    for command in systemctl service; do
-        if which $command &>/dev/null; then
-            echo $command && return
+# This is the Right Way(tm) to check if we are booted with
+# systemd, according to sd_booted(3)
+if [ -d /run/systemd/system ]; then
+    cmd=systemctl
+else
+    # Everything else is really hard to figure out, so we just use
+    # service(8) if it's available - that works with most init
+    # systems/distributions I know of, including FreeBSD
+    if type service >/dev/null 2>&1; then
+        cmd=service
+    else
+        # If even service(8) isn't available, we just try /etc/init.d
+        # and hope for the best
+        if [ -d /etc/init.d ]; then
+            cmd=sysv
+        else
+            echo "Unable to detect a way to restart Jellyfin; bailing out" 1>&2
+            echo "Please report this bug to https://github.com/jellyfin/jellyfin/issues" 1>&2
+            exit 1
         fi
-    done
-    echo "sysv"
-}
+    fi
+fi
+
+if type sudo >/dev/null 2>&1; then
+    sudo_command=sudo
+else
+    sudo_command=
+fi
 
-cmd="$( get_service_command )"
 echo "Detected service control platform '$cmd'; using it to restart Jellyfin..."
 case $cmd in
     'systemctl')
-        echo "sleep 0.5; /usr/bin/sudo $( which systemctl ) start jellyfin" | at now 
+        $sudo_command systemd-run --scope systemctl restart jellyfin
```

Comment:
Wait, this works? As in, it actually starts back up the service properly every time? If so absolutely fantastic.

Reply:
Same as `service`, the `which` would be required here as well to get the full path.

Reply:
> Wait, this works? As in, it actually starts back up the service properly every time? If so absolutely fantastic.

Yep! I mean... at least as far as I can tell :P

`systemd-run` _is_ designed for this though (I think...) so I think it should work all the time. Guess I should probably add a comment explaining why it's needed.

Reply:
Oh that makes perfect sense, my comment was more my surprise that there *was* a way around this - I couldn't find one!